src/OrchardCore.Modules/Orchard.Indexing/IndexingTaskManager.cs

```diff
@@ -138,13 +137,20 @@ public void Dispose()
 
             try
             {
-                var table = $"{_tablePrefix}{nameof(IndexingTask)}";
+                var dialect = SqlDialectFactory.For(connection);
+                var sqlBuilder = dialect.CreateBuilder(_tablePrefix);
+
+                sqlBuilder.Select();
+                sqlBuilder.Table(nameof(IndexingTask));
+                sqlBuilder.Selector("*");
+                sqlBuilder.Take(count);
+                sqlBuilder.WhereAlso($"{dialect.QuoteForColumnName("Id")} > @Id");
 
-                return await connection.QueryAsync<IndexingTask>($"select top {count} * from [{table}] where Id > @Id", new { Id = afterTaskId }, transaction);
+                return await connection.QueryAsync<IndexingTask>(sqlBuilder.ToSqlString(dialect), new { Id = afterTaskId }, transaction);
             }
             catch (Exception e)
             {
-                Logger.LogError("An error occured while reading indexing tasks", e);
+                Logger.LogError("An error occured while reading indexing tasks: " + e.Message);
```

Comment:
Well it wasn't before, if you look at the method definition, first is the message, then the arguments. So I am fixing it here.